src/arb/ArbTypes.jl

```diff
@@ -238,6 +200,55 @@ function _acb_clear_fn(x::acb)
   ccall((:acb_clear, libarb), Nothing, (Ref{acb}, ), x)
 end
 
+################################################################################
+#
+#  Precision management
+#
+################################################################################
+
+struct Balls
+end
+
+# arb as in arblib
+const ARB_DEFAULT_PRECISION = Ref{Int}(64)
+
+function set_precision!(::Type{Balls}, n::Int)
+  arb_check_precision(n)
+  ARB_DEFAULT_PRECISION[] = n
+end
+
+function Base.precision(::Type{Balls})
+  return ARB_DEFAULT_PRECISION[]
+end
+
+function set_precision!(f, ::Type{Balls}, prec::Int)
+  arb_check_precision(prec)
+  old = precision(Balls)
+  set_precision!(Balls, prec)
```

Comment:
So this function is not thread safe (I am not sure if other parts of Nemo are thread safe, though it sure would be nice if they were :-) ).

If `ARB_DEFAULT_PRECISION` is only used by Julia code, I guess this could be rectified (now or in a future update) by making `ARB_DEFAULT_PRECISION` thread local. (E.g. by changing it to a `Vector{Int}`, indexed by `Threads.threadid()`)